### PR TITLE
Add BernoulliLogits distribution

### PR DESCRIPTION
**Pull request by fritzo**
This adds a `BernoulliLogits(logits)` distribution. In the future we may merge this with the Bernoulli distribution, but currently we don't have enough kwargs machinery to handle optional params in distributions.

The motivating use case is a linear-gaussian HMM with a probit likelihood model.

## Tested
- added a unit test for likelihood
- added a unit test for `Bernoulli` as well

### CODE REVIEW

**Comment by fritzo** [comment]
Thanks for reviewing!

> in numpyro ... separate classes for different parametrizations ... is working out well

Interesting. I guess a compromise would be to keep separate classes under the hood (e.g. `BernoulliProbs`, `BernoulliLogits`) and let a top level function dispatch based on available kwargs, e.g.
```py
def Bernoulli(**kwargs):
    """Dispatch to either BernoulliProbs or BernoulliLogits"""
    if 'probs' in kwargs:
        probs = kwargs.pop('probs')
        return BernoulliProbs(probs, **kwargs)
    if 'logits' in kwargs:
        logits = kwargs.pop('logits')
        return BernoulliLogits(logits, **kwargs)
    raise ValueError
```

**Comment by neerajprad** [comment]
I think that's a good suggestion to decouple the interface with the implementation, will make this change.
